### ISSUE TITLE
Respect runtime environment variables

### ISSUE THREAD

**Issue by dargmuesli** [harlan-zw/nuxt-seo-kit#112]
### Describe the bug

Running `nuxt build` and `NUXT_PUBLIC_SITE_URL=https://test.de node .output/server/index.mjs` shows

- `<link rel="canonical" href="http://example.com/">` instead of `<link rel="canonical" href="http://test.de/">`
- `<loc>http://example.com/</loc>` instead of `<loc>http://test.de/</loc>`

Somehow stackblitz throws another error for me (seemingly related to nuxtseo?), but you can download the project and run those commands locally.

### Reproduction

https://stackblitz.com/edit/github-3gjye4-xkhgsi?file=nuxt.config.ts

### System / Nuxt Info

```Shell
- Operating System: Linux
- Node Version:     v20.5.1
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  npm@9.8.0
- Builder:          -
- User Config:      devtools, modules, site
- Runtime Modules:  @nuxtseo/module@2.0.0-beta.33
- Build Modules:    -
```

**Comment by harlan-zw** [comment]
Since this issue was made the logic for nuxt-site-config was improved by quite a bit. It doesn't solve this issue though.

The specific problem here is that you're running your server using environment variables different from your build. Only build time config is considered at the moment.

I'll keep open as ideally it should be supported.

**Comment by harlan-zw** [comment]
Hi, this should be fixed in https://github.com/harlan-zw/nuxt-site-config/releases/tag/v1.6.0

Available in v2.0.0-beta.44

**Comment by dargmuesli** [comment]
Wow, perfect, thank you! :rocket:
